includes/class-wcdn-writepanel.php

```diff
@@ -161,60 +161,75 @@ public function my_bulk_action_handler( $redirect_to, $doaction, $post_ids ) {
 			}
 
 			// security check.
-			if ( isset( $_GET['page'] ) && 'wc-orders' === $_GET['page'] ) {
-				check_admin_referer( 'bulk-orders' );
-			} else {
-				check_admin_referer( 'bulk-posts' );
-			}
-
-			// get referrer.
-			if ( ! wp_get_referer() ) {
-				return $redirect_to;
-			}
+			check_admin_referer( 'bulk-orders' );
 
-			// filter the referer args.
-			$referer_args = array();
-			parse_str( wp_parse_url( wp_get_referer(), PHP_URL_QUERY ), $referer_args );
+			// Get the total number of post IDs.
+			$total     = count( $post_ids );
+			$print_url = htmlspecialchars_decode( wcdn_get_print_link( $post_ids, $template_type ) );
 
-			// set the basic args for the sendback.
-			$args = array();
-			if ( isset( $referer_args['post_status'] ) ) {
-				$args = wp_parse_args( array( 'post_status' => $referer_args['post_status'] ), $args );
-			}
-			if ( isset( $referer_args['paged'] ) ) {
-				$args = wp_parse_args( array( 'paged' => $referer_args['paged'] ), $args );
-			}
-			if ( isset( $referer_args['orderby'] ) ) {
-				$args = wp_parse_args( array( 'orderby' => $referer_args['orderby'] ), $args );
-			}
-			if ( isset( $referer_args['order'] ) ) {
-				$args = wp_parse_args( array( 'orderby' => $referer_args['order'] ), $args );
-			}
-
-			if ( isset( $referer_args['post_type'] ) ) {
-				$args = wp_parse_args( array( 'post_type' => $referer_args['post_type'] ), $args );
-			}
+			// WooCommerce orders page URL.
+			$orders_page_url = admin_url( 'admin.php?page=wc-orders' );
 
-			if ( isset( $referer_args['page'] ) ) {
-				$args = wp_parse_args( array( 'page' => $referer_args['page'] ), $args );
-			}
-
-			// do the action.
-			$total = count( $post_ids );
-			$url   = wcdn_get_print_link( $post_ids, $template_type );
-
-			// generate more args and the sendback string.
-			$args     = wp_parse_args(
-				array(
-					$report_action => true,
-					'total'        => $total,
-					'print_url'    => rawurlencode( $url ),
-				),
-				$args
-			);
-			$sendback = add_query_arg( $args, '' );
-			wp_safe_redirect( $sendback );
-			exit;
+			// Output the modal with Vue.js.
+			?>
+			<div id="custom-modal-app">
+				<div v-if="showModal" class="custom-modal">
+					<div class="custom-modal-content">
+						<!-- Modal Header -->
+						<div class="custom-modal-header">
+						<h1>Print {{ templateType }}</h1>
+							<button class="custom-modal-close" @click="closeModal()">×</button>
+						</div>
+
+						<!-- Modal Body -->
+						<div class="custom-modal-body">
+							<p>Ready to print {{ totalOrders }}  {{ templateType }} .</p>
+						</div>
+
+						<!-- Modal Footer -->
+						<div class="custom-modal-footer">
+							<button class="cancel-button button" @click="closeModal()">Cancel</button>
+							<button class="primary-button button" @click="printDocument()">Print</button>
+						</div>
+					</div>
+				</div>
+			</div>
+			<script>
+				src="https://cdn.jsdelivr.net/npm/vue@2.6.14/dist/vue.js">
```

Comment:
'src' is outside of the script tag and had no opening bracket. It could cause issues.